### DISCUSSION TITLE
why it shows "uninitialized constant FastlaneCore::Helper::LaneHelper (NameError)"

### DISCUSSION THREAD

**Discussion by wade0317** [fastlane/fastlane#17876] [Q&A]
##  Question

When i use Match ,  I write my code such as follow:

              params = create_params_value()
              config = FastlaneCore::Configuration.create(Match::Options.available_options, params)
              Match::Runner.new.run(config)

It was success before version 2.142.0 , but after version 2.142.0,  it  shows  error such as:

	1: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.168.0/match/lib/match/options.rb:258:in `available_options'
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.168.0/match/lib/match/options.rb:15:in `default_platform': uninitialized constant FastlaneCore::Helper::LaneHelper (NameError)

##  Need Help

Why it show error after 2.142.0,  how do i fix my code?  i need use match which code , not command.

## My fastlane env

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 11.1                                                           |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.24.2 (Apple Git-127)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | macOS 11.1 (20C69)                                             |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.4.1                                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.168.0 | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.4.0        |
| slack-notifier            | 2.3.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.7.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.4        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.78.0       |
| ruby2_keywords            | 0.0.2        |
| faraday                   | 1.1.0        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.7        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.2.0        |
| gh_inspector              | 1.1.3        |
| mini_magick               | 4.10.1       |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.6        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| jwt                       | 2.2.2        |
| uber                      | 0.1.0        |
| declarative               | 0.0.20       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.14.1       |
| signet                    | 0.14.0       |
| memoist                   | 0.16.2       |
| os                        | 1.1.0        |
| googleauth                | 0.14.0       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.4.0        |
| google-cloud-errors       | 1.0.1        |
| google-cloud-core         | 1.5.0        |
| rake                      | 13.0.1       |
| digest-crc                | 0.6.1        |
| google-cloud-storage      | 1.29.1       |
| emoji_regex               | 3.2.1        |
| jmespath                  | 1.4.0        |
| aws-partitions            | 1.397.0      |
| aws-eventstream           | 1.1.0        |
| aws-sigv4                 | 1.2.2        |
| aws-sdk-core              | 3.109.3      |
| aws-sdk-kms               | 1.39.0       |
| aws-sdk-s3                | 1.85.0       |
| nanaimo                   | 0.3.0        |
| colored2                  | 3.1.2        |
| claide                    | 1.0.3        |
| CFPropertyList            | 3.0.2        |
| atomos                    | 0.1.3        |
| xcodeproj                 | 1.19.0       |
| json                      | 2.3.1        |
| bundler                   | 2.1.4        |
| forwardable               | 1.3.1        |
| logger                    | 1.3.0        |
| stringio                  | 0.0.2        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| strscan                   | 1.0.0        |
| date                      | 3.0.1        |
| fileutils                 | 1.4.1        |
| io-console                | 0.5.6        |
| zlib                      | 1.0.0        |
| libxml-ruby               | 3.2.1        |
| rexml                     | 3.1.9        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
| webrick                   | 1.4.2        |

**Comment by rogerluan** [comment]
I just tested this same setup locally and it works just fine. This might be related to the fact that you're on macOS BigSur 11.1 (I'm on Catalina). 🤔 

Can you update to latest fastlane version and create a ticket about this [here](https://github.com/fastlane/fastlane/issues/new?assignees=&labels=&template=BUG_REPORT.md)? 🙏
